src/backend/core/tests/test_models_invitations.py

```diff
@@ -73,6 +73,51 @@ def test_models_invitations__is_expired(settings):
     assert expired_invitation.is_expired is True
 
 
+def test_models_invitation__new_user__convert_invitation_to_access():
+    """
+    Upon creating a new identity, invitations linked to that email
+    should be converted to accesses and then deleted.
+    """
+    invitation = factories.InvitationFactory()
+    factories.InvitationFactory(team=invitation.team)  # same team but other email
+    new_identity = factories.IdentityFactory(is_main=True, email=invitation.email)
+
+    assert (
+        models.TeamAccess.objects.filter(
+            team=invitation.team, user=new_identity.user
+        ).exists()
+        is True
+    )
+    assert (
+        len(models.Invitation.objects.filter(team=invitation.team)) == 1
+    )  # the other invitation remains
+
+
+def test_models_invitation__new_user__expired_invitation(settings):
+    """
+    Upon creating a new identity, expired invitations
+    should not be converted into accesses.
+    """
+    invitation = factories.InvitationFactory()
+    settings.INVITATION_VALIDITY_DURATION = 1
+    time.sleep(1)
+
+    new_identity = factories.IdentityFactory(is_main=True, email=invitation.email)
+
+    assert (
+        models.TeamAccess.objects.filter(
+            team=invitation.team, user=new_identity.user
+        ).exists()
+        is False
+    )
+    assert (
+        models.Invitation.objects.filter(
+            team=invitation.team, email=invitation.email
+        ).exists()
+        is True
+    )
```

Comment:
nit: 
```suggestion
    assert not  models.TeamAccess.objects.filter(team=invitation.team, user=new_identity.user).exists()
    assert models.Invitation.objects.filter(team=invitation.team, email=invitation.email).exists()
```
should do the same, I let you lint this suggestion